CONTRIBUTING.md

```diff
@@ -0,0 +1,54 @@
+
+# Contributing
+
+All contributors are expected to follow our [Code of Conduct](CODE_OF_CONDUCT.md).
+
+## Pull requests
+
+Before opening large pull requests, it is prefered that the change be discussed in a github issue first. This helps keep everyone on the same page, and facilitates a smoother code review process.
+
+## Testing
+
+The CI system conducts a few different tests for various releases of rust. In addition to the normal cargo tests, code formatting is checked with [fmt](https://github.com/rust-lang-nursery/rustfmt), and linting is checked with [clippy](https://github.com/rust-lang-nursery/rust-clippy). Whereas cargo tests are run for all rust release channels, `fmt` and `clippy` are only run on the stable channel.
+
+| Channel | fmt | clippy | test |
+|---------|-----|--------|------|
+| stable  | x   | x      | x    |
+| beta    |     |        | x    |
+| nightly |     |        | x    |
+
+Too avoid any surprises by CI while merging, it's recommended you run these locally after making changes. Setup and testing only takes a couple minutes at most.
```

Comment:
nitpick: there's a typo, should be "To avoid..."

Reply:
Ah, good catch! :)